NOTES.WIN

```diff
@@ -93,9 +103,28 @@
    "vcvarsall.bat" before you compile. For example, if you want to build
    "arm64" builds, you should type "vcvarsall.bat x86_arm64 uwp".
 
- mingw and mingw64
+ MinGW and MinGW64
  =================
 
+ In the nutshell MinGW builds are always cross-compiled.
+ On Linux and Cygwin they look exactly as such and require
+ --cross-compile-prefix option. While on MSYS[2] it is solved
+ rather by placing gcc that produces "MinGW binary" code 1st on $PATH.
+ This is customarily source of confusion. "Hosted" applications "live" in
+ emulated file system name space with POSIX-y root, mount points, /dev
+ and even /proc. Confusion is intensified by the fact that MSYS2 shell
+ (or rather emulated execve(2) call) examines the binary it's about to
+ start, and if it's found *not* to be linked with MSYS2 POSIX-y thing,
+ command line arguments that look like file names get translated from
+ emulated name space to "native". For example '/c/some/where' becomes
+ 'c:\some\where', '/dev/null' - 'nul'. This creates an illusion that
+ there is no difference between MSYS2 shell and "MinGW binary", but
+ there is. Just keep in mind that "MinGW binary" "experiences" Windows
+ system in exactly same way as one produced by VC, and in its essence
+ is indistinguishable from the latter. (Which by the way is why
+ it's referred to in quotes here, as "MinGW binary", it's just as
+ "native" as it can get.)
+
```

Comment:
I'd be tempted to move  the section below about building for mingw from Linux/Cygwin into a separate "Cross compiling" section.

Reply:
> I'd be tempted to move the section below about building for mingw from Linux/Cygwin into a separate "Cross compiling" section.

Please check the new MinGW section, which I revised entirely. Ok now?